File: 1038_Binary Search Tree to Greater Sum Tree/BinarySearchTreetoGreaterSumTree_PostOrder.cs
public class Solution {
    private int accumulatedVal = 0;
    public TreeNode BstToGst(TreeNode root) {
        PostOrder(root);
        return root;
    }
    
    private void PostOrder(TreeNode node){
        if(node == null) return;
        PostOrder(node.right);
        node.val += accumulatedVal;
        accumulatedVal = node.val;
        PostOrder(node.left);
    }
}

File: 1038_Binary Search Tree to Greater Sum Tree/BinarySearchTreetoGreaterSumTree_PostOrder.java
class Solution {
    private int accuSum = 0;
    public TreeNode bstToGst(TreeNode root) {
        traverse(root);
        return root;
    }

      private void traverse(TreeNode node) {
        if(node == null) return;
        traverse(node.right);
        int v = node.val;
        node.val += accuSum;
        accuSum += v;
        traverse(node.left);
    }
}